### ISSUE TITLE
Is test split = p1-test + p2-test + p3-test ??

### ISSUE THREAD

**Issue by anirbanl** [frankaging/Reason-SCAN#1]
Hi,

I went through the [code](https://github.com/anirbanl/Reason-SCAN/blob/main/code/dataset/generate_ReaSCAN_splits.ipynb) and found that p1-test, p2-test, p3-test splits are created from the test split of ReaSCAN-v1.0/ReaSCAN-compositional and they when added up should give test split (the numbers also hint that 921+2120+2712 = 5753).

However, when I check the dataset splits (my code snippet below executed in root dir ReaSCAN-v1.0 after extracting ReaSCAN-v1.0.zip), I find that that p1-test+p2-test+p3-test NOT EQ test. Maybe I am doing something wrong here and I needed help in figuring out:

```
import json
import collections
p1=json.load(open('ReaSCAN-compositional-p1-test/data-compositional-splits.txt', 'r'))["examples"]["test"]
p2=json.load(open('ReaSCAN-compositional-p2-test/data-compositional-splits.txt', 'r'))["examples"]["test"]
p3=json.load(open('ReaSCAN-compositional-p3-test/data-compositional-splits.txt', 'r'))["examples"]["test"]
p123=p1+p2+p3
ptest=json.load(open('ReaSCAN-compositional/data-compositional-splits.txt', 'r'))["examples"]["test"]
p123_command=[x['command'] for x in p123]
ptest_command=[x['command'] for x in ptest]
p123_command_sorted = sorted(p123_command)
ptest_command_sorted = sorted(ptest_command)
if collections.Counter(p123_command_sorted) == collections.Counter(ptest_command_sorted):
    print ("The lists are identical")
else :
    print ("The lists are not identical")
```

I get the output: _The lists are not identical_

Further checking produces:
```
>p123_command_sorted[:10]
['pull,the,big,blue,circle',
 'pull,the,big,blue,circle',
 'pull,the,big,blue,circle',
 'pull,the,big,blue,circle',
 'pull,the,big,blue,circle,cautiously',
 'pull,the,big,blue,circle,hesitantly',
 'pull,the,big,blue,circle,that,is,in,the,same,column,as,a,big,blue,square,and,inside,of,a,red,box,while,spinning',
 'pull,the,big,blue,circle,that,is,in,the,same,column,as,a,big,yellow,cylinder,hesitantly',
 'pull,the,big,blue,circle,that,is,in,the,same,column,as,a,big,yellow,cylinder,hesitantly',
 'pull,the,big,blue,circle,that,is,in,the,same,column,as,a,small,blue,square']

>ptest_command_sorted[:10]
['pull,the,big,blue,circle',
 'pull,the,big,blue,circle',
 'pull,the,big,blue,circle',
 'pull,the,big,blue,circle',
 'pull,the,big,blue,circle,cautiously',
 'pull,the,big,blue,circle,cautiously',
 'pull,the,big,blue,circle,hesitantly',
 'pull,the,big,blue,circle,that,is,in,the,same,column,as,a,big,yellow,cylinder,hesitantly',
 'pull,the,big,blue,circle,that,is,in,the,same,column,as,a,small,blue,square,hesitantly',
 'pull,the,big,blue,circle,that,is,in,the,same,column,as,a,small,blue,square,hesitantly']
```

Thanks in advance for your help! :)

**Comment by frankaging** [comment]
Hi Anirban,

Thanks for reaching out, and finding this uploading issue! I think I forgot to update those three splits when I fixed a previous issue on 06/16/2021. Here are two ways you can solve this issue:

(1) You can use the dataset splitter (code snippet attached here) and the main dataset file (ReaSCAN-v1.0/ReaSCAN-compositional/data-compositional-splits.txt) to simply recreate p1-test, p2-test, and p3-test.
p1_test_example_filtered = []
p2_test_example_filtered = []
p3_test_example_filtered = []
for example in data_json["examples"]["test"]:
    if example['derivation'] == "$OBJ_0":
        p1_test_example_filtered += [example]
    elif example['derivation'] == "$OBJ_0 ^ $OBJ_1":
        p2_test_example_filtered += [example]
    elif example['derivation'] == "$OBJ_0 ^ $OBJ_1 & $OBJ_2":
        p3_test_example_filtered += [example]
print(f"p1 test example count={len(p1_test_example_filtered)}")
print(f"p2 test example count={len(p2_test_example_filtered)}")
print(f"p3 test example count={len(p3_test_example_filtered)}")

(2) Or, you can download those updated splits that I just created from this link: https://drive.google.com/file/d/1qelO22kifi_oiWJn2iCuFCsOlqkd3vQp/view?usp=sharing.

Either way should get you the identical splits we used in the paper. I will take care of updating the dataset link with a release note in the next couple of days.

Best,
Zen

**Comment by anirbanl** [comment]
Thank you! This fixes the issue :)

**Comment by anirbanl** [comment]
Hi,

After fixing this issue I find the numbers of #Command-world pairs changes in p1-test, p2-test and p3-test (compared to what is mentioned in Table 3 of the paper - refer GREEN column in my attached image). 

Also I find due to the changed #Command-world pairs, there is a slight difference in the weighted performance of p*-test for M-LSTM (weighted by the updated #Command-world pairs) than what is reported for test split in Table 3, maybe a rounding  error.. although GCN-LSTM matches.. just bringing to your notice if it is of any interest.

![Screenshot 2021-11-26 at 1 01 55 PM](https://user-images.githubusercontent.com/44803404/143544179-1ab9950a-fe68-47bd-8a46-ca7fb510d5cf.png)

Thanks!

**Comment by frankaging** [comment]
Hi, 

Thanks again for bringing up this issue. We provided an updated version of these performance results on our readme file https://github.com/frankaging/Reason-SCAN. 

As noted in the readme file, if you choose to use the old version of testing sets, you want to compare against numbers in the paper. If you use the updated version of testing sets, you want to use the numbers in the readme.

I also updated the dataset to include both versions of testing setss
